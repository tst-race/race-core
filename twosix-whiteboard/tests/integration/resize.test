#!/bin/bash

DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
source "${DIR}"/helpers.sh

start_containers

# TODO health check?
sleep 2
curl --fail --silent --show-error --request GET "localhost:5000/resize/0" | jq -c  'del(.timestamp)'

curl --fail --silent --show-error --request POST "localhost:5000/post/foo" --header 'Content-Type: application/json' --data-raw '{ "data":"1" }' | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request POST "localhost:5000/post/foo" --header 'Content-Type: application/json' --data-raw '{ "data":"2" }' | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request POST "localhost:5000/post/bar" --header 'Content-Type: application/json' --data-raw '{ "data":"3" }' | jq -c  'del(.timestamp)'

curl --fail --silent --show-error --request GET "localhost:5000/resize/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/resize/0" | jq -c  'del(.timestamp)'


echo "get latest index for non-existent category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/latest/foo" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/latest/bar" | jq -c  'del(.timestamp)'

echo "get index for non-existent category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1" | jq -c  'del(.timestamp)'

echo "get range for non-existent category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1/-2" | jq -c  'del(.timestamp)'

echo "post to previously non-existent category" | tee >(cat >&2)
curl --fail --silent --show-error --request POST "localhost:5000/post/foo" --header 'Content-Type: application/json' --data-raw '{ "data":"0" }' | jq -c  'del(.timestamp)'

echo "get latest index for existent category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/latest/foo" | jq -c  'del(.timestamp)'

echo "get latest index for non-existent category when another category exists" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/latest/bar" | jq -c  'del(.timestamp)'

echo "get index for existent category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/3" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1" | jq -c  'del(.timestamp)'

echo "get index for non-existent category when another category exists" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/3" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1" | jq -c  'del(.timestamp)'

echo "get non-existent range for existent category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/-2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/-2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/-2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/3/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/3/-2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/3/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/3/3" | jq -c  'del(.timestamp)'

echo "get existent range for existent category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/3" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/3" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/2/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/2/3" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/2/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/3" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/-1" | jq -c  'del(.timestamp)'

echo "get range for non-existent category when another category exists" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0/3" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1/3" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/2/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/2/3" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/2/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1/3" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1/-1" | jq -c  'del(.timestamp)'

echo "post to existent category" | tee >(cat >&2)
curl --fail --silent --show-error --request POST "localhost:5000/post/foo" --header 'Content-Type: application/json' --data-raw '{ "data":"1" }' | jq -c  'del(.timestamp)'

echo "get latest index for existent category (2)" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/latest/foo" | jq -c  'del(.timestamp)'

echo "get index for existent category (2)" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/3" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/4" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1" | jq -c  'del(.timestamp)'

echo "get non-existent range for existent category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/-3" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/-3" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/-2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/-3" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/2/-3" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/3/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/3/-2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/3/-3" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/4/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/4/3" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/4/4" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/4/-1" | jq -c  'del(.timestamp)'

echo "get existent range for existent category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/3" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/4" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/-2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/3" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/4" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/-2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/2/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/2/3" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/2/4" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/2/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/2/-2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/3/3" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/3/4" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/3/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/3" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/4" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/-1" | jq -c  'del(.timestamp)'

echo "post to second category" | tee >(cat >&2)
curl --fail --silent --show-error --request POST "localhost:5000/post/bar" --header 'Content-Type: application/json' --data-raw '{ "data":"bar0" }' | jq -c  'del(.timestamp)'

echo "get latest index for first category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/latest/foo" | jq -c  'del(.timestamp)'

echo "get index for first category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/3" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/4" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1" | jq -c  'del(.timestamp)'

echo "get latest index for second category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/latest/bar" | jq -c  'del(.timestamp)'

echo "get index for second category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1" | jq -c  'del(.timestamp)'

echo "get non-existent ranges for second category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/2/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/2/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/2/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/2/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1/0" | jq -c  'del(.timestamp)'

echo "get existent ranges for second category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1/-1" | jq -c  'del(.timestamp)'

stop_containers