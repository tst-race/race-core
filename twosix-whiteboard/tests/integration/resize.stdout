building docker image in case it's not up-to-date
create docker network if in case it doesn't exist
start containers
{"mem_usage":0}
{"index":0}
{"index":1}
{"index":0}
{"mem_usage":0}
{"mem_usage":0}
get latest index for non-existent category
{"latest":2}
{"latest":1}
get index for non-existent category
get range for non-existent category
{"data":[],"length":2}
{"data":[],"length":2}
{"data":[],"length":2}
{"data":[],"length":1}
{"data":[],"length":1}
{"data":[],"length":1}
post to previously non-existent category
{"index":2}
get latest index for existent category
{"latest":3}
get latest index for non-existent category when another category exists
{"latest":1}
get index for existent category
{"data":"0"}
{"data":"0"}
get index for non-existent category when another category exists
get non-existent range for existent category
{"data":[],"length":3}
{"data":[],"length":3}
{"data":[],"length":3}
{"data":[],"length":3}
{"data":[],"length":3}
{"data":[],"length":3}
{"data":[],"length":3}
{"data":[],"length":3}
{"data":[],"length":3}
{"data":[],"length":3}
{"data":[],"length":3}
{"data":[],"length":3}
{"data":[],"length":3}
get existent range for existent category
{"data":["0"],"length":3}
{"data":["0"],"length":3}
{"data":["0"],"length":3}
{"data":["0"],"length":3}
{"data":["0"],"length":3}
{"data":["0"],"length":3}
{"data":["0"],"length":3}
{"data":["0"],"length":3}
{"data":["0"],"length":3}
{"data":["0"],"length":3}
{"data":["0"],"length":3}
{"data":["0"],"length":3}
get range for non-existent category when another category exists
{"data":[],"length":1}
{"data":[],"length":1}
{"data":[],"length":1}
{"data":[],"length":1}
{"data":[],"length":1}
{"data":[],"length":1}
{"data":[],"length":1}
{"data":[],"length":1}
{"data":[],"length":1}
{"data":[],"length":1}
{"data":[],"length":1}
{"data":[],"length":1}
{"data":[],"length":1}
{"data":[],"length":1}
{"data":[],"length":1}
{"data":[],"length":1}
{"data":[],"length":1}
{"data":[],"length":1}
post to existent category
{"index":3}
get latest index for existent category (2)
{"latest":4}
get index for existent category (2)
{"data":"0"}
{"data":"1"}
{"data":"1"}
get non-existent range for existent category
{"data":[],"length":4}
{"data":[],"length":4}
{"data":[],"length":4}
{"data":[],"length":4}
{"data":[],"length":4}
{"data":[],"length":4}
{"data":[],"length":4}
{"data":[],"length":4}
{"data":[],"length":4}
{"data":[],"length":4}
{"data":[],"length":4}
{"data":[],"length":4}
{"data":[],"length":4}
{"data":[],"length":4}
{"data":[],"length":4}
{"data":[],"length":4}
{"data":[],"length":4}
{"data":[],"length":4}
{"data":[],"length":4}
get existent range for existent category
{"data":["0"],"length":4}
{"data":["0","1"],"length":4}
{"data":["0","1"],"length":4}
{"data":["0","1"],"length":4}
{"data":["0"],"length":4}
{"data":["0"],"length":4}
{"data":["0","1"],"length":4}
{"data":["0","1"],"length":4}
{"data":["0","1"],"length":4}
{"data":["0"],"length":4}
{"data":["0"],"length":4}
{"data":["0","1"],"length":4}
{"data":["0","1"],"length":4}
{"data":["0","1"],"length":4}
{"data":["0"],"length":4}
{"data":["1"],"length":4}
{"data":["1"],"length":4}
{"data":["1"],"length":4}
{"data":["1"],"length":4}
{"data":["1"],"length":4}
{"data":["1"],"length":4}
post to second category
{"index":1}
get latest index for first category
{"latest":4}
get index for first category
{"data":"0"}
{"data":"1"}
{"data":"1"}
get latest index for second category
{"latest":2}
get index for second category
{"data":"bar0"}
{"data":"bar0"}
get non-existent ranges for second category
{"data":[],"length":2}
{"data":[],"length":2}
{"data":[],"length":2}
{"data":[],"length":2}
{"data":[],"length":2}
{"data":[],"length":2}
{"data":[],"length":2}
get existent ranges for second category
{"data":["bar0"],"length":2}
{"data":["bar0"],"length":2}
{"data":["bar0"],"length":2}
{"data":["bar0"],"length":2}
{"data":["bar0"],"length":2}
{"data":["bar0"],"length":2}
{"data":["bar0"],"length":2}
{"data":["bar0"],"length":2}
{"data":["bar0"],"length":2}
stop containers
